src/overlays/actors/ovl_Boss_Tw/z_boss_tw.c

```diff
@@ -97,242 +119,5033 @@ const ActorInit Boss_Tw_InitVars = {
+                this->work[CS_TIMER_2] = 0;
+                this->subCamYaw = this->subCamYawStep = this->actor.speedXZ = this->subCamDistStep = 0.0f;
+            }
+            break;
+        case 3:
+            BossTw_DeathCSMsgSfx(this, globalCtx);
+            if (this->work[CS_TIMER_2] < 150) {
+                globalCtx->envCtx.unk_BE = 1;
+                globalCtx->envCtx.unk_BD = 0;
+                Math_ApproachZeroF(&globalCtx->envCtx.unk_D8, 1.0f, 0.1f);
+            } else {
+                globalCtx->envCtx.unk_BE = 1;
+                globalCtx->envCtx.unk_BD = 6;
+                Math_ApproachF(&globalCtx->envCtx.unk_D8, (Math_SinS(this->work[CS_TIMER_2] * 4096) / 4.0f) + 0.75f,
+                               1.0f, 0.1f);
+            }
+
+            Math_ApproachF(&this->subCamAt.y, koumePtr->actor.world.pos.y + 17.0f, 0.05f, 10.0f);
+
+            if (this->work[CS_TIMER_2] >= 50) {
+                Math_ApproachF(&this->subCamDist, 110.0f, 0.05f, this->subCamDistStep);
+                Math_ApproachF(&this->subCamDistStep, 1.0f, 1.0f, 0.025f);
+                this->subCamEye.x = this->subCamDist * sinf(this->subCamYaw);
+                this->subCamEye.z = this->subCamDist * cosf(this->subCamYaw);
+                if (this->work[CS_TIMER_2] >= 151) {
+                    this->subCamYaw += this->subCamYawStep;
+                    if (this->work[CS_TIMER_2] >= 800) {
+                        Math_ApproachF(&this->subCamYawStep, 0.0f, 1.0f, 0.0001f);
+                    } else {
+                        Math_ApproachF(&this->subCamYawStep, 0.015f, 1.0f, 0.0001f);
+                    }
+                }
+            }
+            Math_ApproachF(&koumePtr->actor.scale.x, 0.009999999f, 0.1f, 0.001f);
+            Actor_SetScale(&koumePtr->actor, koumePtr->actor.scale.x);
+            Actor_SetScale(&kotakePtr->actor, koumePtr->actor.scale.x);
+            if (this->work[CS_TIMER_2] >= 1020) {
+                Camera* cam = Gameplay_GetCamera(globalCtx, 0);
+                cam->eye = this->subCamEye;
+                cam->eyeNext = this->subCamEye;
+                cam->at = this->subCamAt;
+                func_800C08AC(globalCtx, this->subCamId, 0);
+                this->csState2 = 4;
+                this->subCamId = 0;
+                func_80064534(globalCtx, &globalCtx->csCtx);
+                func_8002DF54(globalCtx, &this->actor, 7);
+                Audio_SetBGM(0x21);
+                Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_DOOR_WARP1, 600.0f, 230.0f,
+                                   0.0f, 0, 0, 0, -1);
+                Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_ITEM_B_HEART, -600.0f, 230.f, 0.0f, 0, 0, 0, 0);
+                this->actor.world.pos.y = -2000.0f;
+                this->workf[UNK_F18] = 0.0f;
+                koumePtr->visible = kotakePtr->visible = false;
+                Flags_SetClear(globalCtx, globalCtx->roomCtx.curRoom.num);
```

Comment:
```suggestion
                koumePtr->visible = kotakePtr->visible = false;
                if (&this->subCamEye) {}
                Flags_SetClear(globalCtx, globalCtx->roomCtx.curRoom.num);
```
this fixes regalloc and there's only stack issues left